cluster-operator/src/test/java/io/strimzi/operator/cluster/model/CruiseControlTest.java

```diff
@@ -0,0 +1,578 @@
+        assertThat(volumeMount, is(notNullValue()));
+        assertThat(volumeMount.getMountPath(), is(CruiseControl.TLS_SIDECAR_CA_CERTS_VOLUME_MOUNT));
+
+        volumeMount = volumesMounts.stream().filter(vol -> CruiseControl.LOG_AND_METRICS_CONFIG_VOLUME_NAME.equals(vol.getName())).findFirst().get();
+        assertThat(volumeMount, is(notNullValue()));
+        assertThat(volumeMount.getMountPath(), is(CruiseControl.LOG_AND_METRICS_CONFIG_VOLUME_MOUNT));
+    }
+
+    @Test
+    public void testEnvVars()   {
+        assertThat(cc.getEnvVars(), is(getExpectedEnvVars()));
+    }
+
+    @Test
+    public void testImagePullPolicy() {
+        Deployment dep = cc.generateDeployment(true, null, ImagePullPolicy.ALWAYS, null);
+        List<Container> containers = dep.getSpec().getTemplate().getSpec().getContainers();
+        Container ccContainer = containers.stream().filter(container -> ccImage.equals(container.getImage())).findFirst().get();
+        assertThat(ccContainer.getImagePullPolicy(), is(ImagePullPolicy.ALWAYS.toString()));
+
+        dep = cc.generateDeployment(true, null,  ImagePullPolicy.IFNOTPRESENT, null);
+        containers = dep.getSpec().getTemplate().getSpec().getContainers();
+        ccContainer = containers.stream().filter(container -> ccImage.equals(container.getImage())).findFirst().get();
+        assertThat(ccContainer.getImagePullPolicy(), is(ImagePullPolicy.IFNOTPRESENT.toString()));
+    }
+
+    @Test
+    public void testContainerTemplateEnvVars() {
+        ContainerEnvVar envVar1 = new ContainerEnvVar();
+        String testEnvOneKey = "TEST_ENV_1";
+        String testEnvOneValue = "test.env.one";
+        envVar1.setName(testEnvOneKey);
+        envVar1.setValue(testEnvOneValue);
+
+        ContainerEnvVar envVar2 = new ContainerEnvVar();
+        String testEnvTwoKey = "TEST_ENV_2";
+        String testEnvTwoValue = "test.env.two";
+        envVar2.setName(testEnvTwoKey);
+        envVar2.setValue(testEnvTwoValue);
+
+        CruiseControlSpec cruiseControlSpec = new CruiseControlSpecBuilder()
+                .withImage(ccImage)
+                .withNewTemplate()
+                    .withNewCruiseControlContainer()
+                       .withEnv(envVar1, envVar2)
+                    .endCruiseControlContainer()
+                .endTemplate()
+                .build();
+
+        Kafka resource = ResourceUtils.createKafkaCluster(namespace, cluster, replicas, image,
+                healthDelay, healthTimeout, metricsCm, kafkaConfig, zooConfig,
+                kafkaStorage, zkStorage, null, kafkaLogJson, zooLogJson, null, cruiseControlSpec);
+        CruiseControl cc = CruiseControl.fromCrd(resource, VERSIONS);
+
+        List<EnvVar> kafkaEnvVars = cc.getEnvVars();
+        assertThat(kafkaEnvVars.stream().filter(var -> testEnvOneKey.equals(var.getName())).map(EnvVar::getValue).findFirst().get(), is(testEnvOneValue));
```

Comment:
`assertThat(kafkaEnvVars, hasEntry(testEnvOneKey, testEnvOneValue)` suggested improvement
And maybe have same for envar2?

Reply:
See above